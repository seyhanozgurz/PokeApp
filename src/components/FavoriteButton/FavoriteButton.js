import {TouchableOpacity, Image, Text} from 'react-native';
import {styles} from './button.styles';
import {RedHeart, Heart} from '../../icons';
function FavoriteButton({onFavorite, isFavorite}) {
  return (
    <TouchableOpacity testID="button" onPress={onFavorite}>
      <Image
        style={styles.button}
        source={isFavorite ? RedHeart : Heart}></Image>
    </TouchableOpacity>
  );
}
export default FavoriteButton;
