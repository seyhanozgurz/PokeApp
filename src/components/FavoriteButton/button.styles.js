import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  button: {
    height: 40,
    width: 40,
    marginTop: 130,
  },
  
});
