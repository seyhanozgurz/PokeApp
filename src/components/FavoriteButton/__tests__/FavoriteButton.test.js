import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import FavoriteButton from '../FavoriteButton';

describe('FavoriteButton', () => {
  it('responds to press events', () => {
    const handlePress = jest.fn();
    const {getByTestId, toJSON} = render(
      <FavoriteButton onFavorite={handlePress} isFavorite={false} />,
    );
    const button = getByTestId('button');

    fireEvent.press(button);
    expect(handlePress).toHaveBeenCalled();
    expect(toJSON()).toMatchSnapshot();
  });
});
