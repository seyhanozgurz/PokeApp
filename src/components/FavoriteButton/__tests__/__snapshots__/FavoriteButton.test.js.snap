// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton responds to press events 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="button"
>
  <Image
    source={
      {
        "testUri": "../../../src/icons/heartIcon.png",
      }
    }
    style={
      {
        "height": 40,
        "marginTop": 130,
        "width": 40,
      }
    }
  />
</View>
`;
